oci-library/oci-hol/java-management/set-up-logrotate/set-up-logrotate.md

```diff
@@ -0,0 +1,138 @@
+
+### Objectives
+
+In this lab, you will:
+
+* Set up logrotate on Linux instances
+* Set up logrotatewin on Windows instances
+
+### Prerequisites
+
+* Access to your host machine with management agent already setup
+* Completed labs 1 to 4
+
+## Task 1: Set up logrotate on Linux instances
+
+1. Run the following two commands:
+
+    ```
+    <copy>
+    sudo yum install logrotate
+    sudo touch /etc/logrotate.d/jms
+    </copy>
+    ```
+
+2. Open the file /etc/logrotate.d/jms in a text editor: 
+
+    ```
+    <copy>
+    sudo nano /etc/logrotate.d/jms
+    </copy>
+    ```
+
+3. Paste the following into the file:
+
+    ```
+    <copy>
+    /var/log/java/usagetracker.log {
+      su mgmt_agent root
+      create 666 mgmt_agent root
+      rotate 5
+      size 20M
+      notifempty
+      compress
+      missingok
+    }
+    </copy>
+    ```
+
+4. To save the file, press CTRL+x. Before exiting, nano will ask you if you wish to save the file: Type y to save and exit, type n to abandon your changes and exit.
+
+## Task 2: Set up logrotatewin on Windows instances
+
+Logrotatewin is a Windows implementation of the logrotate utility found in Linux platforms. The goal is to use the same command line parameters and files as the Linux version. It is not provided by JMS and you will need to install it separately.
+
+1. Download logrotatewin from [https://sourceforge.net/projects/logrotatewin/files/](https://sourceforge.net/projects/logrotatewin/files/)
+
+2. Extract the downloaded zip file and run `logrotateSetup` to start the installer. In this example, we changed the install destination to `C:\Program Files\LogRotate\`
+    * ![image of logrotatewin install destination](/../images/logrotate-installation-example.png)
+    * The installer will install to the path `C:\Program Files\LogRotate`
+    * The executable file will be in `C:\Program Files\LogRotate\logrotate.exe`
+    * The configuration file will be in `C:\Program Files\LogRotate\Content\logrotate.conf`
+
+
+3. We want to setup logrotate for `usagetracker.log` file: 
+    * Determine the path to the `usagetracker.log` on your machine by looking for the `com.oracle.usagetracker.logToFile` parameter in the Java Usage Tracker config file.
+    * The Java Usage Tracker config file is located by default in `C:\Program Files\Java\conf\usagetracker.properties`
+    * Right-click on the config file and open with notepad to inspect it:
+    * ![image of a sample usagetracker.properties file](/../images/usagetracker-properties-example.png)
+
+4. Once you have identified the path of `usagetracker.log`, edit `C:\Program Files\LogRotate\Content\logrotate.conf` using notepad:
+    * Login as an **Administrator** and open a Command Prompt window
+        * To check if you are currently running the Command Prompt as an Administrator, enter the following:
+        ```
+        <copy>
+        ECHO Administrative permissions required for installation of management agent. Detecting permissions...
```

Comment:
This portion need to add the first 2 lines because if the system doesn't find NET SESSION, error level will be 9009 (some feedback in the previous drafts). So if it isn't set up as the path variable in the system, it will always indicate permissions are not sufficient even if it is already administrator.

SETLOCAL
PATH=C:\Windows\system32;%PATH%
ECHO Administrative permissions required for installation of management agent. Detecting permissions...
NET SESSION >nul 2>&1
IF %ERRORLEVEL% == 0 (
  ECHO Success: Administrative permissions detected.
) ELSE (
  ECHO Failure: Current permissions insufficient. Please reopen the Command Prompt with administrative permissions.
)

Reply:
good catch, will try it out and amend it. @xinyitay  fyi on above  ^

Reply:
ok seems like SETLOCAL is only needed if used in a batch file. For copy pasting into terminal it is not required.

Nonetheless no harm leaving it in, in case people blindly copy into a batch file